app/lib/frontend/templates/views/shared/search_tabs.dart

```diff
@@ -2,32 +2,43 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'package:json_annotation/json_annotation.dart';
+import '../../../dom/dom.dart' as d;
+import '../../../static_files.dart';
 
-part 'search_tabs.g.dart';
-
-@JsonSerializable(createFactory: false, explicitToJson: true)
-class SearchTabs {
-  final List<SearchTab>? tabs;
-
-  SearchTabs({this.tabs});
-
-  Map<String, dynamic> toJson() => _$SearchTabsToJson(this);
-}
-
-@JsonSerializable(createFactory: false, explicitToJson: true)
 class SearchTab {
-  final bool? active;
-  final String? href;
-  final String? title;
-  final String? text;
+  final bool active;
+  final String href;
+  final String title;
+  final String text;
 
   SearchTab({
-    this.active,
-    this.href,
-    this.title,
-    this.text,
+    required this.active,
+    required this.href,
+    required this.title,
+    required this.text,
   });
+}
 
-  Map<String, dynamic> toJson() => _$SearchTabToJson(this);
+String renderSearchTabs(Iterable<SearchTab> tabs) {
```

Comment:
Your suggested pattern is that in general things that produce `Node` exist inside a `views/*.dart` file...

But that in cases where we have a data-class we add the render methods next to the data-class.

I don't particularly mind either way, I think a good thing about `dom.dart` is that we can _go-to definition_ on all templates :D

--------

**Side note**:

Couldn't `SearchTab` just be a function that returns `Node`? or are we actually caching these somewhere, just curious?

Even if we do... it's entirely possible that it would just as fine to just cache the `Node` object.. not that we have to refactor this now -- it's probably best to do these things step by step.

I'm just wondering if in some cases, the intermediate data-class might just be functions that return `Node` which can be cached.

Reply:
Maybe we could have a `Node` constructor that just takes the cached HTML output and keeps it, without parsing, emitting when needed. It could be used only in the caching layer...

Let's see how the template migration goes and we can figure out a good pattern for it.